mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -379,15 +379,21 @@ class RowMatrix @Since("1.0.0") (
    *
    * Note that this cannot be computed on matrices with more than 65535 columns.
    *
-   * @param k number of top principal components.
+   * @param filter either the number of top principal components or variance
+   *               retained by the minimal set of principal components.
    * @return a matrix of size n-by-k, whose columns are principal components, and
    * a vector of values which indicate how much variance each principal component
    * explains
    */
   @Since("1.6.0")
-  def computePrincipalComponentsAndExplainedVariance(k: Int): (Matrix, Vector) = {
+  def computePrincipalComponentsAndExplainedVariance(filter: Either[Int, Double])
```

Comment:
@sethah @jodersky It looks like the comment Since("1.6.0") is false becaue this method is not available in spark 1.6 - this change was merged to master instead of 1.6 branch. Do you still consider this change as API breaking given that it modifies API that wasn't yet released? If yes then I'll do as @jodersky said and introduce a new method and move common code to a new private one. I'd really like to have this feature in MLlib version because I use it.